src/simplicial-hott/05-segal-types.rzk.md

```diff
@@ -216,6 +216,36 @@ Extension types are also used to define the type of commutative triangles:
         t₂ ≡ t₁ ↦ h t₂]   -- the diagonal is exactly `h`
 ```
 
+## Free arrow types
+
+A free arrow in `A` can be described as a triple consisting of two endpoints and
+the arrow between them. This is essentially a special case of
+`cofibration-composition`, but invoking it would require passing through the
+identification `(product A A) ≃ (∂Δ¹ → A)`. Hence it is faster to just show it
+directly.
+
+```rzk
+#def free-arrows
+  ( A : U)
+  : U
+  :=
+    Σ ((a,b) : product A A), hom A a b
+
+#def free-arrow
+  ( A : U)
+  : ( Δ¹ → A) → free-arrows A
+  := \ σ → ((σ 0₂, σ 1₂), σ)
+
+#def is-equiv-free-arrow
+  ( A : U)
+  : is-equiv (Δ¹ → A) (free-arrows A) (free-arrow A)
+  :=
+    ( ( \ ((_,_), σ) → σ
+      , \ _ → refl)
+    , ( \ ((_,_), σ) → σ
+      , \ _ → refl))
```

Comment:
This is hard to parse like this. Generally, lambdas as subexpressions should be parenthesized, and the arguments of comma pairings, when the total expression spans multiple lines. Lastly, commas should have whitespace before and after:
```suggestion
    ( ( ( \ ((_ , _) , σ) → σ)
      , ( \ _ → refl))
    , ( ( \ ((_ , _) , σ) → σ)
      , ( \ _ → refl)))
```

Reply:
The parentheses are redundant. They are only redundant in the sense that `rzk` will parse the code and accept it as valid proof irrespective of whether the parentheses are there or not. However, they add uniformity to the definition, and this way the reader does not have to trust that the author formatted the proof correctly to parse it correctly. This is similar to how we parenthesize single-token arguments in multi-line function applications even though that is also "redundant".

E.g.
```rzk
my-function
  ( single-token-argument-1)
  ( some more complicated expression)
```

Reply:
Another valid option for formatting this proof is
```rzk
    ( ( ( \ ((_ , _) , σ) → σ) , (\ _ → refl))
    , ( ( \ ((_ , _) , σ) → σ) , (\ _ → refl)))
```

Reply:
Ok, thanks!